types/field_type.go

```diff
@@ -1258,6 +1262,49 @@ var fieldTypeMergeRules = [fieldTypeNum][fieldTypeNum]byte{
 	},
 }
 
+const (
+	InvalidResult byte = 0 /** not valid for UDFs */
+	StringResult  byte = 1 /** char * */
+	RealResult    byte = 2 /** double */
+	IntResult     byte = 3 /** long long */
+	RowResult     byte = 4 /** not valid for UDFs */
+	DecimalResult byte = 5 /** char *, to be converted to/from a decimal */
+)
+
+var fieldTypesResultType = []byte{
+	// mysql.TypeDecimal      mysql.TypeTiny
+	DecimalResult, IntResult,
+	// mysql.TypeShort        mysql.TypeLong
+	IntResult, IntResult,
+	// mysql.TypeFloat        mysql.TypeDouble
+	RealResult, RealResult,
+	// mysql.TypeNull         mysql.TypeTimestamp
+	StringResult, StringResult,
+	// mysql.TypeLonglong     mysql.TypeInt24
+	IntResult, IntResult,
+	// mysql.TypeDate         mysql.TypeTime
+	StringResult, StringResult,
+	// mysql.TypeDateTime     mysql.TypeYear
+	StringResult, IntResult,
+	// mysql.TypeNewDate      mysql.TypeVarchar
+	StringResult, StringResult,
+	// mysql.TypeBit          mysql.TypeInvalid
+	IntResult, InvalidResult,
+	// Unused entries: <17>-<242>
```

Comment:
In previous definitions, the matrix order is TypeBit, <16>-<244>, TypeJSON, ...

In this implementation, the order is TypeBit, TypeInvalid, <17>-<242>, TypeBool, TypeJSON

Will there be problems?